### PR TITLE
[chore][ncl] Update contacts screen

### PR DESCRIPTION
**Pull request by EvanBacon**
# Why

- moved from #8857 
- migrate contacts screen to functional components
- configure for react-navigation v5

# Test Plan

- test in simulators
Screen Shot 2020-06-17 at 8 27 59 PM

### CODE REVIEW

File: apps/native-component-list/src/components/HeaderIconButton.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+export function HeaderContainerRight(props: any) {
+  return <View {...props} style={[{ paddingRight: 8, flexDirection: 'row' }, props.style]} />;
+}
+
+export default function HeaderIconButton({
+  onPress,
+  size = 24,
+  name,
+  color = 'blue',
```
Comment (ide) [inline apps/native-component-list/src/components/HeaderIconButton.tsx:13]: sort these properties the same as you did in the type definition right below

File: apps/native-component-list/src/screens/Contacts/ContactsScreen.tsx [hunk 2]
```diff
@@ -1,78 +1,66 @@
...
-    const { status } = await Permissions.askAsync(Permissions.CONTACTS);
-    this.setState({ permission: status === 'granted' });
-  }
+  return <ContactsView navigation={navigation} />;
+}
 
-  loadAsync = async ({ distanceFromEnd }: { distanceFromEnd?: number } = {}, restart = false) => {
-    if (!this.state.permission || this.state.refreshing) {
+function ContactsView({ navigation }: { navigation: any /* StackNavigationProp<StackParams> */ }) {
+  let _rawContacts: Record<string, Contacts.Contact> = {};
```
Comment (ide) [inline apps/native-component-list/src/screens/Contacts/ContactsScreen.tsx:39]:
```suggestion
  let rawContacts: Record<string, Contacts.Contact> = {};
```

File: apps/native-component-list/src/screens/Contacts/ContactsScreen.tsx [hunk 3]
```diff
@@ -1,78 +1,66 @@
...
+
+  const [contacts, setContacts] = React.useState<Contacts.Contact[]>([]);
+  const [hasPreviousPage, setHasPreviousPage] = React.useState<boolean>(true);
```
Comment (ide) [inline apps/native-component-list/src/screens/Contacts/ContactsScreen.tsx:42]: The inferred type should be sufficient:
```suggestion
  const [hasPreviousPage, setHasPreviousPage] = React.useState(true);
```

File: apps/native-component-list/src/screens/Contacts/ContactsScreen.tsx [hunk 4]
```diff
@@ -1,78 +1,66 @@
...
+  const [hasNextPage, setHasNextPage] = React.useState<boolean>(true);
+  const [refreshing, setRefreshing] = React.useState<boolean>(false);
+
+  const onPressItem = React.useCallback(
+    (id: string) => {
+      // tslint:disable-next-line no-console
```
Comment (ide) [inline apps/native-component-list/src/screens/Contacts/ContactsScreen.tsx:48]: remove?

File: apps/native-component-list/src/screens/Contacts/ContactsScreen.tsx [hunk 5]
```diff
@@ -83,66 +71,43 @@ export default class ContactsScreen extends React.Component<NavigationScreenProp
...
-
-  onPressItem = async (id: string) => {
-    // tslint:disable-next-line no-console
-    console.log('onPress', id);
-    this.props.navigation.navigate('ContactDetail', { id });
-  }
+    setContacts(Object.values(_rawContacts));
+    setHasPreviousPage(payload.hasPreviousPage);
+    setHasNextPage(payload.hasNextPage);
+    setRefreshing(false);
```
Comment (ide) [inline apps/native-component-list/src/screens/Contacts/ContactsScreen.tsx:86]: These aren't going to be batched IIRC (four renders)

File: apps/native-component-list/src/screens/Contacts/ContactsScreen.tsx [hunk 6]
```diff
@@ -152,3 +117,33 @@ const styles = StyleSheet.create({
...
   },
 });
+
+function AddContactHeaderButton() {
+  return (
+    <HeaderIconButton
+      name="md-add"
+      onPress={async () => {
+        const randomContact = { note: 'Likes expo...' };
+        // @ts-ignore
```
Comment (ide) [inline apps/native-component-list/src/screens/Contacts/ContactsScreen.tsx:127]:
```suggestion
        // @ts-ignore: why?
```

File: apps/native-component-list/src/screens/Contacts/ContactsScreen.tsx [hunk 7]
```diff
@@ -152,3 +117,33 @@ const styles = StyleSheet.create({
...
+        ContactUtils.presentNewContactFormAsync({ contact: randomContact });
+        // ContactUtils.presentUnknownContactFormAsync({
+        //   contact: randomContact,
+        // });
```
Comment (ide) [inline apps/native-component-list/src/screens/Contacts/ContactsScreen.tsx:131]: remove

File: apps/native-component-list/src/screens/Contacts/ContactDetailScreen.tsx [hunk 8]
```diff
@@ -1,168 +1,149 @@
...
-        iconSize={23}
-        color="blue">
-        <HeaderButtons.Item
-          title="share"
-          iconName="md-share"
+ContactDetailScreen.navigationOptions = ({
+  navigation,
+}: // route: {
+//   params: { id },
+// },
```
Comment (ide) [inline apps/native-component-list/src/screens/Contacts/ContactDetailScreen.tsx:46]: delete?

File: apps/native-component-list/src/screens/Contacts/ContactDetailScreen.tsx [hunk 9]
```diff
@@ -1,168 +1,149 @@
...
+});
 
-  checkPermissionAsync = async () => {
-    const permission = await getPermissionAsync(Permissions.CONTACTS);
-    this.setState({ permission });
+function ContactDetailView({
+  navigation,
+}: // route: {
+//   params: { id },
+// },
```
Comment (ide) [inline apps/native-component-list/src/screens/Contacts/ContactDetailScreen.tsx:74]: rm

File: apps/native-component-list/src/screens/Contacts/ContactsScreen.tsx (hunk 5 above)
Comment (EvanBacon) [reply apps/native-component-list/src/screens/Contacts/ContactsScreen.tsx]: I got rid of one

tenor

File: apps/native-component-list/src/screens/Contacts/ContactsScreen.tsx [hunk 10]
```diff
@@ -74,19 +59,17 @@ function ContactsView({ navigation }: { navigation: any /* StackNavigationProp<S
...
-      _rawContacts = {};
+      rawContacts = {};
     }
 
     for (const contact of nextContacts) {
-      _rawContacts[contact.id] = contact;
+      rawContacts[contact.id] = contact;
     }
-    setContacts(Object.values(_rawContacts));
-    setHasPreviousPage(payload.hasPreviousPage);
```
Comment (ide) [inline apps/native-component-list/src/screens/Contacts/ContactsScreen.tsx]: is it safe to just remove this? seems like it'd break a feature

Comment (EvanBacon) [reply apps/native-component-list/src/screens/Contacts/ContactsScreen.tsx]: We weren't using the result of it.

**Pull request merged**